components/gfx/font_context.rs

```diff
@@ -160,76 +103,88 @@ impl<S: FontSource> FontContext<S> {
         font_group
     }
 
-    /// Returns a reference to an existing font cache entry matching `descriptor` and `family`, if
-    /// there is one.
-    fn font_cache_entry(&self, descriptor: &FontDescriptor, family: &SingleFontFamily) -> Option<&FontCacheEntry> {
-        self.font_cache.iter()
-            .find(|cache_entry| cache_entry.matches(&descriptor, &family))
-    }
-
-    /// Creates a new font cache entry matching `descriptor` and `family`.
-    fn create_font_cache_entry(&mut self, descriptor: &FontDescriptor, family: &SingleFontFamily) -> FontCacheEntry {
-        let font =
-            self.font_source.find_font_template(family.clone(), descriptor.template_descriptor.clone())
-                .and_then(|template_info|
-                    self.create_font(template_info, descriptor.to_owned()).ok()
-                )
-                .map(|font| Rc::new(RefCell::new(font)));
-
-        FontCacheEntry { family: family.atom().to_owned(), font }
-    }
-
-    /// Returns a font from `family` matching the `descriptor`. Fonts are cached, so repeated calls
-    /// will return a reference to the same underlying `Font`.
-    pub fn font(&mut self, descriptor: &FontDescriptor, family: &SingleFontFamily) -> Option<FontRef> {
-        if let Some(entry) = self.font_cache_entry(descriptor, family) {
-            return entry.font.clone()
-        }
-
-        let entry = self.create_font_cache_entry(descriptor, family);
-        let font = entry.font.clone();
-        self.font_cache.push(entry);
-        font
-    }
+    /// Returns a font matching the parameters. Fonts are cached, so repeated calls will return a
+    /// reference to the same underlying `Font`.
+    pub fn font(
+        &mut self,
+        font_descriptor: &FontDescriptor,
+        family_descriptor: &FontFamilyDescriptor,
+    ) -> Option<FontRef>
+    {
```

Comment:
nit: brace to the previous line.

Reply:
Thanks, I thought this was the style [recommended here](https://github.com/rust-lang-nursery/fmt-rfcs/blob/master/guide/items.md) but I've realised they only do a line break when there is a `where` clause.